### PR TITLE
feat: add name field to tasks bucket to match cloud

### LINKED ISSUE #23156: Unify _tasks bucket schema in OSS and Cloud
Ref: https://github.com/influxdata/EAR/issues/2960

### PR DESCRIPTION
**Pull request by jeffreyssmith2nd**
Adds missing field to tasks bucket for compatibility with Cloud.

Closes #23156 

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: task/backend/analytical_storage.go [hunk 1]
```diff
@@ -34,7 +39,7 @@ const (
 // RunRecorder is a type which records runs into an influxdb
 // backed storage mechanism
 type RunRecorder interface {
-	Record(ctx context.Context, orgID platform.ID, org string, bucketID platform.ID, bucket string, run *taskmodel.Run) error
+	Record(ctx context.Context, bucketID platform.ID, bucket string, task *taskmodel.Task, run *taskmodel.Run) error
```
Comment (DStrand1) [inline task/backend/analytical_storage.go:42]: Is this a change that requires a change to OpenAPI?

File: task/backend/run_recorder.go [hunk 2]
```diff
@@ -67,5 +71,5 @@ func (s *StoragePointsWriterRecorder) Record(ctx context.Context, orgID platform
 	}
 
 	// TODO - fix
```
Comment (DStrand1) [inline task/backend/run_recorder.go:73]: Any clues on what this TODO is for, if it was referring to this new addition or something else?

**Review by DStrand1 — APPROVED** [review]
LGTM! Just a couple minor comments

File: task/backend/analytical_storage.go (hunk 1 above)
Comment (jeffreyssmith2nd) [reply task/backend/analytical_storage.go:42]: I don't think so, I haven't found anywhere that this would be directly exposed as an API.

**Pull request merged**